sdk/core/azure-core/azure/core/rest/_aiohttp.py

```diff
@@ -23,14 +23,62 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
-
+import collections.abc
 import asyncio
+from itertools import groupby
 from typing import AsyncIterator
 from multidict import CIMultiDict
 from . import HttpRequest, AsyncHttpResponse
 from ._helpers_py3 import iter_raw_helper, iter_bytes_helper
 from ..pipeline.transport._aiohttp import AioHttpStreamDownloadGenerator
 
+class _ItemsView(collections.abc.ItemsView):
+    def __init__(self, ref):
+        super().__init__(ref)
+        self._ref = ref
+
+    def __iter__(self):
+        for key, groups in groupby(self._ref.__iter__(), lambda x: x[0]):
+            yield tuple([key, ", ".join(group[1] for group in groups)])
+
+    def __contains__(self, item):
+        assert isinstance(item, (list, tuple))
+        assert len(item) == 2
+        for k, v in self.__iter__():
+            if item[0] == k and item[1] == v:
+                return True
+        return False
+
+    def __repr__(self):
+        return f"dict_items({list(self.__iter__())})"
+
+
+class _CIMultiDict(CIMultiDict):
+    """Dictionary with the support for duplicate case-insensitive keys."""
+
+    def __iter__(self):
+        return iter(self.keys())
+
+    def keys(self):
+        """Return a new view of the dictionary's keys."""
+        return {k: v for k, v in self.items()}.keys()  # pylint: disable=unnecessary-comprehension
+
+    def items(self):
+        """Return a new view of the dictionary's keys."""
+        return _ItemsView(super().items())
+
+    def values(self):
+        """Return a new view of the dictionary's values."""
+        return {k: v for k, v in self.items()}.values()  # pylint: disable=unnecessary-comprehension
```

Comment:
Does is need to be a comprehension? I think you can just do `dict(self.items()).values()`? (And can remove the pylint msg)

Reply:
...and @annatisch's comment applies to `keys()` as well....

Reply:
yay like that a lot better, thanks!